mobile/src/main/java/org/openhab/habdroid/model/LinkedPage.java

```diff
@@ -75,10 +78,12 @@ public static LinkedPage fromJson(JSONObject jsonObject) {
         if (jsonObject == null) {
             return null;
         }
+        String icon = jsonObject.optString("icon", null);
         return new AutoValue_LinkedPage.Builder()
                 .id(jsonObject.optString("id", null))
                 .title(jsonObject.optString("title", null))
-                .icon(jsonObject.optString("icon", null))
+                .icon(icon)
+                .iconPath(String.format("icon/%s", icon))
```

Comment:
The format differs for openHAB 1 and 2. This is why I put it here.

Reply:
Makes sense ... however, looking at the format, it seems like the cloud notification code uses the OH1 format ('images/%s.png') as well ... I wonder whether we need to do something there; and if we do, I wonder *what* to do there (no ServerProperties at that point).

Reply:
TBH I don't really like that api at all. The app must create a socket to the server in order to download the icon and display it in the notification. This is discouraged by Google and I'm afraid a slow connection will cause the app to be killed before the notification is shown.
I haven't found time so far to write an issue, but what about putting the icon inside the FCM message, even if that means that we might have to drop png support for those icons (due to size restrictions)?

Reply:
> This is discouraged by Google and I'm afraid a slow connection will cause the app to be killed before the notification is shown.

Why do you think that's the case? `FcmMessageListenerService` ultimately derives from `IntentService`, which executes in a background thread.

Do you mean that quote?
```
Because a small portion of the Android mobile population are on high latency networks, avoid opening a connection to your servers before displaying a notification. Calling back to the server before the end of the allowed processing time may be risky for users on high latency networks. 
```
If so, to comply to that, we could look into posting the notification without icon at first and then, after downloading the icon, update it to include the icon.

> but what about putting the icon inside the FCM message, even if that means that we might have to drop png support for those icons (due to size restrictions)?

Not sure whether openhab-cloud people will be fond of storing the icons in their database...and if they don't do that, the very same question applies to showing the list of (previous) notifications.